src/pages/workspace/taxes/WorkspaceTaxCodePage.tsx

```diff
@@ -0,0 +1,108 @@
+import type {StackScreenProps} from '@react-navigation/stack';
+import React, {useCallback} from 'react';
+import {View} from 'react-native';
+import {useOnyx} from 'react-native-onyx';
+import FormProvider from '@components/Form/FormProvider';
+import InputWrapper from '@components/Form/InputWrapper';
+import type {FormOnyxValues} from '@components/Form/types';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import ScreenWrapper from '@components/ScreenWrapper';
+import TextInput from '@components/TextInput';
+import useAutoFocusInput from '@hooks/useAutoFocusInput';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import {setPolicyTaxCode, validateTaxCode} from '@libs/actions/TaxRate';
+import Navigation from '@libs/Navigation/Navigation';
+import type {SettingsNavigatorParamList} from '@libs/Navigation/types';
+import AccessOrNotFoundWrapper from '@pages/workspace/AccessOrNotFoundWrapper';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import type SCREENS from '@src/SCREENS';
+import INPUT_IDS from '@src/types/form/WorkspaceTaxCodeForm';
+
+type WorkspaceTaxCodePageProps = StackScreenProps<SettingsNavigatorParamList, typeof SCREENS.WORKSPACE.TAX_CODE>;
+
+function WorkspaceTaxCodePage({route}: WorkspaceTaxCodePageProps) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+    const policyID = route.params.policyID ?? '-1';
+    const currentTaxCode = route.params.taxID;
+
+    const [policy] = useOnyx(`${ONYXKEYS.COLLECTION.POLICY}${policyID}`);
+    const {inputCallbackRef} = useAutoFocusInput();
+
+    const setTaxCode = useCallback(
+        (values: FormOnyxValues<typeof ONYXKEYS.FORMS.WORKSPACE_TAX_CODE_FORM>) => {
+            const newTaxCode = values.taxCode.trim();
+            if (currentTaxCode === newTaxCode) {
+                Navigation.goBack(ROUTES.WORKSPACE_TAX_EDIT.getRoute(policyID, currentTaxCode));
+                return;
+            }
+
+            setPolicyTaxCode(policyID, currentTaxCode, newTaxCode);
+            Navigation.dismissModal();
+            Navigation.navigate(ROUTES.WORKSPACE_TAX_EDIT.getRoute(policyID, newTaxCode));
+        },
+        [currentTaxCode, policyID],
+    );
+
+    const validate = useCallback(
+        (values: FormOnyxValues<typeof ONYXKEYS.FORMS.WORKSPACE_TAX_CODE_FORM>) => {
+            if (!values[INPUT_IDS.TAX_CODE]) {
+                return {};
+            }
+            if (!policy) {
+                return {};
+            }
```

Comment:
I think we don't need this check either. The page will throw not found if there's no policy accessible from `AccessOrNotFoundWrapper`.